backup/moodle2/restore_learningmap_activity_task.class.php

```diff
@@ -99,25 +100,29 @@ public function after_restore(): void {
         }
         $oldmapid = $placestore->mapid;
         $newmapid = uniqid();
-        $item->intro = str_replace('learningmap-svgmap-' . $oldmapid, 'learningmap-svgmap-' . $newmapid, $item->intro);
         $placestore->mapid = $newmapid;
 
-        if (!isset($placestore->version) || $placestore->version < 2024022102) {
-            $placestore->version = 2024022102;
+        if (!isset($placestore->version) || $placestore->version < 2024032402) {
+            $placestore->version = 2024032402;
             // Needs 1 as default value (otherwise all place strokes would be hidden).
             if (!isset($placestore->strokeopacity)) {
                 $placestore->strokeopacity = 1;
             }
-            $mapworker = new \mod_learningmap\mapworker($item->intro, (array)$placestore);
+            if (empty($item->svgcode)) {
+                $mapcode = $item->intro;
+                $item->intro = '';
+                $item->showmaponcoursepage = $cm->showdescription;
+            } else {
+                $mapcode = $item->svgcode;
+            }
+            $mapworker = new \mod_learningmap\mapworker($mapcode, (array)$placestore);
```

Comment:
When you backup in the old version and restore in the new, the filearea is "intro" and not "background".